dist/s_test_suite_no_executable

```diff
@@ -1,9 +1,12 @@
 #!/usr/bin/env bash
 
-cd ../test/suite
-if [[ $(find . -executable -type f | grep -v -w "./run.py") ]]; then
-    echo "The following files should not be executable: " &&
-    find . -executable -type f | grep -v -w "./run.py"
+# All test scripts should be routed through "run.py", running them directly is an anti-pattern.
+# Use the executable bits on the scripts as weak evidence of the anti-pattern.
+workspace=$(git rev-parse --show-toplevel)
+exes=$(find $workspace/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) -a ! -name 'run.py' -print)
```

Comment:
what's the purpose of the singular -o?

Reply:
That's shorthand for `-or` (same as how the `-a` after the parentheses is `-and`).

Reply:
`-a` is implied and can be removed.

Reply:
Removed